sdk/storage/AzureStorageBlob/Source/DataStructures/StorageBlobClientOptions.swift

```diff
@@ -47,7 +47,7 @@ public struct StorageBlobClientOptions: AzureConfigurable {
     public init(
         apiVersion: String = StorageBlobClient.ApiVersion.latest.rawValue,
         logger: ClientLogger = ClientLoggers.default(tag: "StorageBlobClient"),
-        maxChunkSize: Int = 4 * 1024 * 1024
+        maxChunkSize: Int = 4 * 1024 * 1024 - 1
```

Comment:
Above this value, you will get a 400 error.

Reply:
Oh, we should definitely document that in the parameter comment. We should also validate this in the StorageBlobClient initializer and throw an exception (since those initializers are already marked as throwing).

Reply:
I added the parameter comment, but adding the validation in the SDK was always an anti-pattern in CLI. If the Storage service were updated to allow hashes greater than this limit, we would be artificially restricting the user. The requirements to use hashing are clearly spelled out in the REST API docs:
https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob#request-headers

Reply:
OH, I see it's only a limitation for requests that use hashing. Yep, what you wrote makes perfect sense.